[luafilesystem] Header file `lfs.h` not found

### Operating system

Windows 11 Pro

### Compiler

Microsoft (R) C/C++ Optimizing Compiler Version 19.41.34120 for x64

### Steps to reproduce the behavior

`vcpkg.json`:

```
{
    "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg/master/scripts/vcpkg.schema.json",
    "name": "helloworld",
    "version-semver": "0.1.0",
    "builtin-baseline": "68d349964cb4e8da561fd849d9491e6ba11c5681",
    "dependencies": [
        {
            "name": "lua",
            "version>=": "5.4.6"
        },
        {
            "name": "luafilesystem",
            "version>=": "1.8.0#6"
        }
    ]
}
```

---

`CMakeLists.txt`:
```cmake
cmake_minimum_required(VERSION 3.27)

project("helloworld" VERSION "0.1.0" LANGUAGES C CXX RC)


find_package(lua "5.4.6" REQUIRED)
find_package(unofficial-luafilesystem REQUIRED CONFIG)


add_executable(helloworld_main main.cpp)

target_include_directories(
	helloworld_main
	PRIVATE ${LUA_INCLUDE_DIR}
)

target_link_libraries(
  helloworld_main
  PRIVATE ${LUA_LIBRARIES} unofficial::luafilesystem::lfs
)
```

---

`main.cpp`:
```cpp
#include <iostream>
#include <lfs.h>

auto main(int argc, char* argv[]) -> int {
    std::cout << "Hello world!\n";
    return 0;
}
```

---

Now build and observe that `lfs.h` is not found.


### Failure logs

```
e:\Projects\helloworld> cmake . -B out/build -DCMAKE_PROJECT_TOP_LEVEL_INCLUDES=C:\apps\vcpkg\scripts\buildsystems\vcpkg.cmake -DVCPKG_APPLOCAL_DEPS=OFF -DVCPKG_TARGET_TRIPLET=x64-windows-static -G Ninja -DCMAKE_BUILD_TYPE=Release; cmake --build out/build --config Release
-- Running vcpkg install
Detecting compiler hash for triplet x64-windows...
Compiler found: C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.41.34120/bin/Hostx64/x64/cl.exe
Detecting compiler hash for triplet x64-windows-static...
Compiler found: C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.41.34120/bin/Hostx64/x64/cl.exe
The following packages will be built and installed:
    lua:x64-windows-static@5.4.7 -- C:\Apps\vcpkg\buildtrees\versioning_\versions\lua\97783a4c337df419c0a6a75e9599545c54259d86
    luafilesystem:x64-windows-static@1.8.0#6 -- C:\Apps\vcpkg\buildtrees\versioning_\versions\luafilesystem\4b474bdcc3f49eef949ba79ad3294556e39af778
  * vcpkg-cmake:x64-windows@2024-04-23 -- C:\Apps\vcpkg\buildtrees\versioning_\versions\vcpkg-cmake\e74aa1e8f93278a8e71372f1fa08c3df420eb840
  * vcpkg-cmake-config:x64-windows@2024-05-23 -- C:\Apps\vcpkg\buildtrees\versioning_\versions\vcpkg-cmake-config\97a63e4bc1a17422ffe4eff71da53b4b561a7841
Additional packages (*) will be modified to complete this operation.
Restored 4 package(s) from C:\Users\shodan\AppData\Local\vcpkg\archives in 57 ms. Use --debug to see more details.
Installing 1/4 vcpkg-cmake-config:x64-windows@2024-05-23...
Elapsed time to handle vcpkg-cmake-config:x64-windows: 5.39 ms
vcpkg-cmake-config:x64-windows package ABI: 2bcf8687e11bbf4f2505fe945e5582e3289a7162661c0513146764ab14d63f77
Installing 2/4 vcpkg-cmake:x64-windows@2024-04-23...
Elapsed time to handle vcpkg-cmake:x64-windows: 5.86 ms
vcpkg-cmake:x64-windows package ABI: cd17b2ece6f2b02dda28e3dd194208eb5776a923163bfb49bba8d0d188310465
Installing 3/4 lua:x64-windows-static@5.4.7...
Elapsed time to handle lua:x64-windows-static: 13.1 ms
lua:x64-windows-static package ABI: be605c7c23c8d4d390c23a6ce7d7876fb93b42bae64b658d5ad2e9ef6bb03766
Installing 4/4 luafilesystem:x64-windows-static@1.8.0#6...
Elapsed time to handle luafilesystem:x64-windows-static: 7.54 ms
luafilesystem:x64-windows-static package ABI: 86289648afe4fddcd827b977940e89928af5fd684c4901e5471ebe3738befd78
Total install time: 32 ms
Use this package via the module FindLua that comes with CMake. To use in your CMakeLists.txt:

    find_package(Lua REQUIRED)
    target_include_directories(main PRIVATE ${LUA_INCLUDE_DIR})
    target_link_libraries(main PRIVATE ${LUA_LIBRARIES})

luafilesystem provides CMake targets:

  find_package(unofficial-luafilesystem CONFIG REQUIRED)
  target_link_libraries(main PRIVATE unofficial::luafilesystem::lfs)

-- Running vcpkg install - done
-- The C compiler identification is MSVC 19.41.34120.0
-- The CXX compiler identification is MSVC 19.41.34120.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.41.34120/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.41.34120/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (4.5s)
-- Generating done (0.0s)
-- Build files have been written to: E:/Projects/helloworld/out/build
[1/2] Building CXX object CMakeFiles\helloworld_main.dir\main.cpp.obj
FAILED: CMakeFiles/helloworld_main.dir/main.cpp.obj
C:\PROGRA~1\MICROS~2\2022\PROFES~1\VC\Tools\MSVC\1441~1.341\bin\Hostx64\x64\cl.exe  /nologo /TP  -external:IE:\Projects\helloworld\out\build\vcpkg_installed\x64-windows-static\include -external:W0 /DWIN32 /D_WINDOWS /EHsc /O2 /Ob2 /DNDEBUG -MD /showIncludes /FoCMakeFiles\helloworld_main.dir\main.cpp.obj /FdCMakeFiles\helloworld_main.dir\ /FS -c E:\Projects\helloworld\main.cpp
E:\Projects\helloworld\main.cpp(3): fatal error C1083: Cannot open include file: 'lfs.h': No such file or directory
ninja: build stopped: subcommand failed.
```

### Additional context

`find_package` can find the lib, but does not know about its version string. i.e. specifying the version with `find_package(unofficial-luafilesystem "1.8.0" REQUIRED CONFIG)` will fail.

--- a/ports/luafilesystem/CMakeLists.txt
+++ b/ports/luafilesystem/CMakeLists.txt
@@ -10,6 +10,7 @@ add_library(lfs src/lfs.h src/lfs.c src/lfs.def)
 
 target_include_directories(lfs PRIVATE ${LFS_INCLUDES})
 target_link_libraries(lfs PRIVATE ${LFS_LIBRARIES})
+target_include_directories(lfs INTERFACE $<INSTALL_INTERFACE:include/luafilesystem>) 
 
 install(TARGETS lfs
     EXPORT "unofficial-${PROJECT_NAME}-targets"
@@ -31,6 +32,8 @@ write_basic_package_version_file(
   VERSION       "${LFS_VERSION}"
   COMPATIBILITY SameMajorVersion
 )
+
+install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/src/lfs.h" DESTINATION "include/luafilesystem")
 
 install(
   FILES "${CMAKE_CURRENT_BINARY_DIR}/unofficial-${PROJECT_NAME}-config.cmake"

--- a/ports/luafilesystem/portfile.cmake
+++ b/ports/luafilesystem/portfile.cmake
@@ -29,8 +29,5 @@ file(REMOVE_RECURSE
 )
 
 # Handle copyright
-file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
 file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
-
-# Allow empty include directory
-set(VCPKG_POLICY_EMPTY_INCLUDE_FOLDER enabled)

--- a/ports/luafilesystem/vcpkg.json
+++ b/ports/luafilesystem/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "luafilesystem",
   "version": "1.8.0",
-  "port-version": 6,
+  "port-version": 7,
   "description": "LuaFileSystem is a Lua library developed to complement the set of functions related to file systems offered by the standard Lua distribution.",
   "homepage": "https://github.com/keplerproject/luafilesystem",
   "supports": "!uwp",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5602,7 +5602,7 @@
     },
     "luafilesystem": {
       "baseline": "1.8.0",
-      "port-version": 6
+      "port-version": 7
     },
     "luajit": {
       "baseline": "2023-01-04",

--- a/versions/l-/luafilesystem.json
+++ b/versions/l-/luafilesystem.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "00b7638338af5a3a2d95c3c9b1ed870ed0cfb9fc",
+      "version": "1.8.0",
+      "port-version": 7
+    },
+    {
       "git-tree": "4b474bdcc3f49eef949ba79ad3294556e39af778",
       "version": "1.8.0",
       "port-version": 6

